File: client/src/app/services/timer/timer.service.ts
import { Injectable } from '@angular/core';
import { NUMERIC_FORMAT_THRESHOLD, TIME_FORMAT_THRESHOLD } from './timer.service.constants';

@Injectable({
    providedIn: 'root',
})
export class TimerService {
    private sec: number;
    private min: number;

    constructor() {
        this.sec = 0;
        this.min = 0;
    }

    updateTime(time: number) {
        this.min = Math.floor(time / TIME_FORMAT_THRESHOLD);
        this.sec = time - this.min * TIME_FORMAT_THRESHOLD;
    }

    formatTime() {
        return this.formatNumber(this.min) + ':' + this.formatNumber(this.sec);
    }

    formatNumber(number: number) {
        return number < NUMERIC_FORMAT_THRESHOLD ? '0' + number : number;
    }
}

File: light_client/lib/timer/timer-service.dart
import 'package:flutter/material.dart';
import 'package:namer_app/general-classes/time-constants.dart';

class TimerService extends ChangeNotifier {
  static final TimerService _instance = TimerService._internal();
  late int _sec;
  late int _min;

  factory TimerService() {
    return _instance;
  }

  TimerService._internal() {
    _sec = 0;
    _min = 0;
  }

  void updateTime(int time) {
    _min = (time / TIME_FORMAT_THRESHOLD).floor();
    _sec = time - _min * TIME_FORMAT_THRESHOLD;
    notifyListeners();
  }

  String formatNumber(int number) {
    return number < NUMERIC_FORMAT_THRESHOLD ? '0$number' : number.toString();
  }

  String formatTime() {
    return '${formatNumber(_min)}:${formatNumber(_sec)}';
  }
}